Rearranging ADO page (#6826)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/git/setup-azure.md

sidebar_label: "Set up Azure DevOps"
---

# Set up Azure DevOps <Lifecycle status="enterprise" />

## Service principal overview

:::note

> The instructions in this section are for creating a service principal app. This feature is being slowly rolled out to dbt Cloud accounts. If you need to configure dbt Cloud and Azure DevOps but don't see the **Service principal** option in your account settings, use the [**Service user**](#set-up-service-user) configuration for now. You can easily [migrate to a service principal](#migrate-to-service-principal) when it is available for your account.

Reviewer: ```suggestion
You can use these instructions to create a service principal app. The service principal feature is being gradually rolled out to dbt Cloud accounts, so it may not be available in all accounts. 

If you need to configure dbt Cloud and Azure DevOps but don't see the **Service principal** option in your account settings, use the [**Service user**](#set-up-service-user) configuration for now. You can easily [migrate to a service principal](#migrate-to-service-principal) when available for your account.
```

:::

> To use dbt Cloud's native integration with Azure DevOps, an account admin needs to set up an Microsoft Entra ID app as a service principal. We recommend setting up a separate [Entra ID application than used for SSO](/docs/cloud/manage-access/set-up-sso-microsoft-entra-id).

Author: ```suggestion
To use dbt Cloud's native integration with Azure DevOps, an account admin needs to set up a Microsoft Entra ID app as a service principal. We recommend setting up a separate [Entra ID application than used for SSO](/docs/cloud/manage-access/set-up-sso-microsoft-entra-id).
```

> To use dbt Cloud's native integration with Azure DevOps, an account admin needs to set up an Microsoft Entra ID app as a service principal. We recommend setting up a separate [Entra ID application than used for SSO](/docs/cloud/manage-access/set-up-sso-microsoft-entra-id).

Reviewer: ```suggestion
To use dbt Cloud's native integration with Azure DevOps, an account admin needs to set up a Microsoft Entra ID app as a service principal. We recommend setting up a separate [Entra ID application than used for SSO](/docs/cloud/manage-access/set-up-sso-microsoft-entra-id).
```

> The application's service principal represents the Entra ID application object. Whereas a service user represents a real user in Azure with an Entra ID (and an applicable license), the service principal is a secure identity used by an application to access Azure resources unattended. The service principal authenticates with a client ID and secret rather than a username and password (or any other form of user auth). Service principals are the [Microsoft recommended method](https://learn.microsoft.com/en-us/entra/architecture/secure-service-accounts#types-of-microsoft-entra-service-accounts) for authenticating apps. 

Reviewer: ```suggestion
The application's service principal represents the Entra ID application object. While a "service user" represents a real user in Azure with an Entra ID (and an applicable license), the "service principal" is a secure identity used by an application to access Azure resources unattended. The service principal authenticates with a client ID and secret rather than a username and password (or any other form of user auth). Service principals are the [Microsoft recommended method](https://learn.microsoft.com/en-us/entra/architecture/secure-service-accounts#types-of-microsoft-entra-service-accounts) for authenticating apps. 
```

sidebar_label: "Set up Azure DevOps"
---

# Set up Azure DevOps <Lifecycle status="enterprise" />

## Service principal overview

:::note

You can use these instructions to create a service principal app. The service principal feature is being gradually rolled out to dbt Cloud accounts, so it may not be available in all accounts. 

> If you need to configure dbt Cloud and Azure DevOps but don't see the **Service principal** option in your account settings, use the [**Service user**](#set-up-service-user) configuration for now. You can easily [migrate to a service principal](#migrate-to-service-principal) when available for your account.

Reviewer 2: This URL for service user doesn't work. I think it needs to be `#connect-a-service-user`

> If you need to configure dbt Cloud and Azure DevOps but don't see the **Service principal** option in your account settings, use the [**Service user**](#set-up-service-user) configuration for now. You can easily [migrate to a service principal](#migrate-to-service-principal) when available for your account.

Author: ```suggestion
If you need to configure dbt Cloud and Azure DevOps but don't see the **Service principal** option in your account settings, use the [**Service user**](#service-user-overview) configuration for now. You can easily [migrate to a service principal](#migrate-to-service-principal) when available for your account.
```

sidebar_label: "Set up Azure DevOps"
---

# Set up Azure DevOps <Lifecycle status="enterprise" />

## Service principal overview

:::note

> 

Author: This is new

sidebar_label: "Set up Azure DevOps"
---

# Set up Azure DevOps <Lifecycle status="enterprise" />

## Service principal overview

:::note

> These instructions are for creating a service principal app. This feature is slowly being rolled out to accounts. If you need to configure dbt Cloud and Azure DevOps but don't see the **service principal** option in your account settings, use the [**service user**](#set-up-service-user) option for now. You can easily [migrate to a service principal](#migrate-to-service-principal) when it is available for your account.

Author: ```suggestion
The instructions in this section are for creating a service principal app. This feature is being slowly rolled out to dbt Cloud accounts. If you need to configure dbt Cloud and Azure DevOps but don't see the **Service principal** option in your account settings, use the [**Service user**](#set-up-service-user) configuration for now. You can easily [migrate to a service principal](#migrate-to-service-principal) when it is available for your account.
```
